### DISCUSSION TITLE
Fetching app sources fails for Azure DevOps on-prem git repo

### DISCUSSION THREAD

**Discussion by DanielGreguric** [codemagic-ci-cd/codemagic-docs#2769] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Set Repository access settings for Azure DevOps Server (On-Prem) git repo with domain user and pass
2. Run any workflow and see "Fetch app sources" step fails with "The requested URL returned error: 500"

**Note, Azure DevOps Services (cloud version) has "Generate Git Credentials" option on git repo, while Azure DevOps Server (on-prem) doesn't. I was able to fetch sources normally from the cloud version, but can't get it to work from on-prem.**

### Expected results

I expect that codemagic pulls the git repo from our Azure DevOps Server (on-prem) project

### Actual results

Fetching app sources
== Cloning repository https://our.org.dom/tfs/DefaultCollection/ADOProject/_git/ADOGITREPO for main ==
> /usr/bin/git clone --progress --branch main --depth 50 https://myusername@our.org.dom/tfs/DefaultCollection/ADOProject/_git/ADOGITREPO /Users/builder/clone
Cloning into '/Users/builder/clone'...
fatal: unable to access 'https://our.org.dom/tfs/DefaultCollection/ADOProject/_git/ADOGITREPO/': The requested URL returned error: 500

Build failed :|
Failed to clone repository

### Build id (optional)

66df0ae903c3609f072ad5d0

**Comment by dam-ease** [comment]
Hello @DanielGreguric.
The buildID shared seems to be for a build over 30 days ago as I am unable to view logs for this build. Are you still running into this issue?
If yes, kindly share the buildID to a recent build with this error.
Also, if this is a privately hosted repository please confirm that you have whitelisted IP addresses from:
https://docs.codemagic.io/getting-started/adding-apps/#firewall-configuration-for-privately-hosted-repositories

**Comment by KPOINT-Team** [comment]
Hi @dam-ease , We are also facing similar issue - We have git repo hosted on AWS Codecommit. Here is the build [link ](https://codemagic.io/app/680b366c427b7056247f8027/build/680b8d9dc0825214d45e8693)for checking logs.
Can you help please ?

**Reply by KPOINT-Team** [reply]
I switched to ssh login and it worked.
